### PR TITLE
Feature/dropdownMenu

### PR DESCRIPTION
**Pull request by Plofland**
## PR TITLE

**Description:** 
Sorry for the length of this PR, it got a bit out of hand.

The purpose of this PR is to put the discussion drawer header and discussion dropdown menu into their own components.

To do this I moved the flagging and moderator modal state to the redux store and all components using these slices of state have been updated.

The depreciated components (DCardDropdown & PopOverContent) were deleted along with their corresponding state in the redux store.

To avoid merge conflicts with Ismail's work on the comment system, I pulled down from the main and resolved the conflicts locally.
This may have still caused some problems because I'm not seeing added comments pop up in the discussionDrawer.
  
  
  
  

### Tests
Ran tests locally.
  

### Checklist

 * [x] My code follows the style guidelines of this project
 * [x] I have performed a self-review of my own code
 * [ ] My code has been reviewed by at least one peer
 * [x] I have commented my code, particularly in hard-to-understand areas
 * [ ] I have made corresponding changes to the documentation
 * [x] My changes generate no new warnings
 * [ ] I have added tests that prove my fix is effective or that my feature works
 * [x] New and existing unit tests pass locally with my changes
 * [ ] There are no merge conflicts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,10 +24,11 @@
     "react-hot-toast": "^1.0.1",
     "react-icons": "^4.2.0",
     "react-modal": "^3.12.1",
-    "react-redux": "^7.2.0",
+    "react-redux": "^7.2.3",
     "react-router-dom": "^5.2.0",
     "react-scripts": "^4.0.3",
     "redux": "^4.0.5",
+    "redux-logger": "^3.0.6",
```
Comment (ialkamal) [inline package.json]: Add redux logger as a dev dependency. 
`npm uninstall <package-name>`
and install as dev dependency
`npm install <package-name> --save-dev`

File: src/assets/DiscussionCard.js [hunk 2]
```diff
@@ -34,14 +35,14 @@ const IconText = ({ icon, text }) => (
...
 
-  const flagsLength = props.discussion.flags
-    ? props.discussion.flags.length
-    : 0;
+  // I believe these lines of code can be deleted if there is code elsewhere handling the discussion flags
+  // const { roomID } = useParams();
+
+  // const flagsLength = props.discussion.flags
+  //   ? props.discussion.flags.length
+  //   : 0;
```
Comment (ialkamal) [inline src/assets/DiscussionCard.js]: yes and the useEffect also as these are being handled by the Feed component

**Review by robo-chick — APPROVED** [review]
Nice work!

**Review by ialkamal — CHANGES_REQUESTED** [review]
Since showing comments is critical, I would wait till you fix the issue.

File: package.json (hunk 1 above)
Comment (Plofland) [reply package.json]: AH! Thank you. I moved it there because I was having issues with the redux logger but I never moved it back once it was fixed.

**Pull request closed without merging**